### PR TITLE
Fix for http client call to wireserver

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:
This PR fixes 3 issues:
1. http get call non blocking
2. check for http status code and return error on failure
3. fix logging issue in ipam

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: ipam/azure.go [hunk 1]
```diff
@@ -84,14 +85,23 @@ func (s *azureSource) refresh() error {
...
 	if err != nil {
+		log.Printf("[ipam] wireserver call failed with: %v", err)
 		return err
 	}
 
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		log.Printf("[ipam] http return error code for wireserver call %d", resp.StatusCode)
+		return fmt.Errorf("http error %d", resp.StatusCode)
```
Comment (ashvindeodhar) [inline ipam/azure.go]: you can log resp.Status instead. This logs code and what that code means.

File: cni/ipam/plugin/main.go [hunk 2]
```diff
@@ -20,7 +25,16 @@ func main() {
...
 
-	ipamPlugin, err := ipam.NewPlugin(&config)
+	log.SetName(name)
+	log.SetLevel(log.LevelInfo)
+	if err := log.SetTarget(log.TargetLogfile); err != nil {
+		fmt.Printf("Failed to setup cni logging: %v\n", err)
+		return
+	}
+
+	defer log.Close()
```
[the remark this answers is not in the document]
Comment (tamilmani1989) [reply cni/ipam/plugin/main.go]: nope it should be called only if log.SetTarget succeeded

File: ipam/azure.go [hunk 3]
```diff
@@ -84,14 +85,23 @@ func (s *azureSource) refresh() error {
 		return err
 	}
 
+	httpClient := &http.Client{Timeout: time.Second * 5}
```
[the remark this answers is not in the document]
Comment (tamilmani1989) [reply ipam/azure.go]: as discussed offline, you already did this in you PR..you will change once this PR is merged to master

**Pull request merged**